src/components/input.tsx

```diff
@@ -0,0 +1,39 @@
+import { useFormContext } from "react-hook-form";
+
+interface InputProps {
+  id: string;
+  placeholder: string;
+}
+
+const Input: React.FC<InputProps> = ({ id, placeholder }) => {
```

Comment:
tutaj raczej nie ma potrzeby destrukturyzacji możemy wziąć props i przekazać je bezpośrednio do inputa:

_<input
      {...props}
        className="w-full rounded-[5px] bg-backgrounds-loginInput h-[40px] px-[16px] py-[10px]"
      />_

jeśli przyjdą w przyszłości inne wartości które trzeba będzie przekazać do innych elementów niż input to wtedy będzie można je restrukturyzować { someProp, ...props}
dzięki temu tez nie musimy korzystać z useFormContext z tego co widzę i error message przekazywać z propsów

```
{ errorMessage, ...inputProps }) => {
.....
return (
    <div className="relative">
      <input
       {...props}
        className="w-full rounded-[5px] bg-backgrounds-loginInput h-[40px] px-[16px] py-[10px]"
      />
      {errors[id] ? (
        <span className="text-[10px] text-[#EA1818] absolute left-0 bottom-[-15px]">
          {errorMessage}
        </span>
      ) : (
        ""
      )}
    </div>
  );

```

Reply:
Jak w tym przypadku użyjemy `register(id)`?

Reply:
Jak w tym przypadku użyte będzie `register(id)`?

Reply:
to zostawmy jeszcze destrukturyzację id i przekażmy register(id)